api/src/DataPersister/PeriodDataPersister.php

```diff
@@ -19,9 +22,56 @@ public function __construct(
         );
     }
 
+    public function beforeCreate($data): BaseEntity {
+        static::managePeriodDays($data);
+
+        return $data;
+    }
+
+    public function beforeUpdate($data): BaseEntity {
+        $orig = $this->em->getUnitOfWork()->getOriginalEntityData($data);
+
+        static::managePeriodDays($data, $orig);
+
+        return $data;
+    }
+
     public function beforeRemove($data): ?BaseEntity {
         $this->validator->validate($data, ['groups' => ['delete', 'Period:delete']]);
 
         return null;
     }
+
+    public static function managePeriodDays(Period $period, array $orig = null) {
+        $length = $period->getPeriodLength();
+        $days = $period->getDays();
+
+        $minDelta = 0;
+        if (null != $orig) {
+            $minDelta = $orig['start']->getTimestamp() - $period->start->getTimestamp();
+            $minDelta = floor($minDelta / 60);
+        }
+
+        // Add Days
+        $i = count($days);
+        while ($i < $length) {
+            $day = new Day();
+            $day->dayOffset = $i++;
```

Comment:
Wenn ich das Startdatum einer Periode vorverschiebe, und das Enddatum am gleichen Datum belasse, werden die Days inklusive DayResponsibles vorverschoben, und am Ende des Lagers werden neue, leere Days eingefügt. Könnte etwas unintuitiv für API-Clients sein die sich nur genau auf diese Operation fokussieren. Aber ich glaube ist schon gut so, wenn man das big picture ansieht. Wäre ja irgendwie auch komisch wenn wir Spezial-Logik einführen würden, die nur dann greift wenn die alle vormaligen Tage noch in der neuen Datums-Range drin sind.

Reply:
Eigentlich würde es Sinn machen, das Verhalten von `moveScheduleEntries` abhängig zu machen:
Wenn `moveScheduleEntries=true`, dann wäre das jetzige Verhalten erwartet. ScheduleEntries + DayResponsibles werden verschoben und am Ende gibt es neue Days.

Wenn `moveScheduleEntries=false` ist (z.B. ich möchte die Anreise am Tag vor dem Lager noch hinzufügen, alles andere bleibt aber gleich), dann wäre es eigentlich logischer, wenn die DayResponsibles stehen bleiben, und am Anfang ein neuer Tag hinzugefügt wird.

Reply:
Guter Punkt!
Spätestens wenn wir Tages-Jobs implementieren werden wir wieder an dieser Frage vorbei kommen.
Ich sehe hier folgende Möglichkeiten:
- Tages-Jobs werden immer mit dem Start-Datum mit geschoben 
   (so ist es aktuell; da an Day-Objekt angehängt; aber evtl. verwirrend)
- Tages-Jobs bleiben immer am gleichen Datum
- Benutzer-Eingabe für ScheduleEntries auch für Jobs verwenden  (mein Favorit)
- Zusätzliche Eingabe von Benutzer verlangen

Reply:
Dann müssten wir bei `moveScheduleEntries=false` zusätzlich auch validieren, dass keine Tagesverantwortlichen auf Tagen stehen, die gelöscht werden.

Reply:
Müssten wir wahrscheinlich sowieso einführen, unabhänig von dieser Diskussion: Ich kann jetzt aktuell eine Periode kürzen, ohne das Startdatum zu ändern. Dann werden am Ende die Days gelöscht, incl. DayResponsibles. 

Oder wir entscheiden uns dafür, dass der Informationsverlust in diesem Fall gering ist und wir erlauben löschen von Tagen, selbst wenn da noch DayReponsibles drauf sind.

Reply:
Das fände ich eher seltsam, wenn man vorher alle Job-Zuteilungen manuell löschen muss, bevor man die Periode verschieben kann.
Da würde ich mich spontan eher für eine Warnung aussprechen.

Reply:
Good for me. Allenfalls müssen wir die Präsentation im Frontend anpassen um bewusster zu kommunizieren, was gerade passiert. Anstatt frei Start und End-datum editieren zu können (aktuelle Implementation) könnten wir die z.B. die folgenden Operationen anbieten:

- Start-Datum anpassen // Default; request wird mit moveScheduleEntires=true abgesetzt
- x Tage am Anfang oder Start hinzufügen // Benutzer kann die Anzahl Tage auswählen; request wird mit moveScheduleEntires=false abgesetzt
- x Tage am Anfang oder am Start löschen // fette Warnung, dass damit möglicherweise Inhalt gelöscht wird; Benutzer kann die Anzahl Tage auswählen; request wird mit moveScheduleEntries=false abgesetzt

Je nach dem, was man erreichen möchte, braucht es dann mehrere Requests. Ist dafür expliziter.